### PR TITLE
Sync rustc_codegen_cranelift

### PR DESCRIPTION
**Pull request by bjorn3**
This time Cranelift has been updated to 0.98. A couple of bugs have been fixed and a decent amount of x86 vendor intrinsics have been implemented.

r? @ghost

@rustbot label +A-codegen +A-cranelift +T-compiler

### CODE REVIEW

File: example/mini_core.rs [hunk 1]
```diff
@@ -563,11 +564,6 @@ unsafe fn allocate(size: usize, _align: usize) -> *mut u8 {
     libc::malloc(size)
 }
 
-#[lang = "box_free"]
-unsafe fn box_free<T: ?Sized>(ptr: Unique<T>, _alloc: ()) {
```
Comment (vi) [inline example/mini_core.rs]: Shall stale comments like [this](https://github.com/rust-lang/rust/blob/098d4fd74c078b12bfc2e9438a2a04bc18b393bc/library/alloc/src/boxed.rs#L195) be updated after this removal?

Comment (bjorn3) [reply example/mini_core.rs]: Should probably have been done in https://github.com/rust-lang/rust/pull/100036 which did the actual removal of `box_free`, but yes it should be updated.

Comment (bjorn3) [reply example/mini_core.rs]: Opened https://github.com/rust-lang/rust/pull/120372

**Pull request merged**